drivers/kw41zrf/kw41zrf_netdev.c

```diff
@@ -0,0 +1,1248 @@
+            *((netopt_state_t *)value) = kw41zrf_netdev_get_state(dev);
+            return sizeof(netopt_state_t);
+
+        case NETOPT_MAX_PACKET_SIZE:
+            assert(len >= sizeof(uint16_t));
+            *((uint16_t *)value) = KW41ZRF_MAX_PKT_LENGTH - _MAX_MHR_OVERHEAD;
+            return sizeof(uint16_t);
+
+        case NETOPT_PRELOADING:
+            assert(len >= sizeof(netopt_enable_t));
+            *((netopt_enable_t *)value) =
+                !!(dev->flags & KW41ZRF_OPT_PRELOADING);
+            return sizeof(netopt_enable_t);
+
+        case NETOPT_PROMISCUOUSMODE:
+            assert(len >= sizeof(netopt_enable_t));
+            *((netopt_enable_t *)value) =
+                !!(dev->flags & KW41ZRF_OPT_PROMISCUOUS);
+            return sizeof(netopt_enable_t);
+
+        case NETOPT_RX_START_IRQ:
+            assert(len >= sizeof(netopt_enable_t));
+            *((netopt_enable_t *)value) =
+                !!(dev->flags & KW41ZRF_OPT_TELL_RX_START);
+            return sizeof(netopt_enable_t);
+
+        case NETOPT_RX_END_IRQ:
+            assert(len >= sizeof(netopt_enable_t));
+            *((netopt_enable_t *)value) =
+                !!(dev->flags & KW41ZRF_OPT_TELL_RX_END);
+            return sizeof(netopt_enable_t);
+
+        case NETOPT_TX_START_IRQ:
+            assert(len >= sizeof(netopt_enable_t));
+            *((netopt_enable_t *)value) =
+                !!(dev->flags & KW41ZRF_OPT_TELL_TX_START);
+            return sizeof(netopt_enable_t);
+
+        case NETOPT_TX_END_IRQ:
+            assert(len >= sizeof(netopt_enable_t));
+            *((netopt_enable_t *)value) =
+                !!(dev->flags & KW41ZRF_OPT_TELL_TX_END);
+            return sizeof(netopt_enable_t);
+
+        case NETOPT_CSMA:
+            assert(len >= sizeof(netopt_enable_t));
+            *((netopt_enable_t *)value) =
+                !!(dev->flags & KW41ZRF_OPT_CSMA);
+            return sizeof(netopt_enable_t);
+
+        case NETOPT_CSMA_RETRIES:
+            assert(len >= sizeof(uint8_t));
+            *((uint8_t *)value) = dev->csma_max_backoffs;
+            return sizeof(uint8_t);
+
+        case NETOPT_CSMA_MAXBE:
+            assert(len >= sizeof(uint8_t));
+            *((uint8_t *)value) = dev->csma_max_be;
+            return sizeof(uint8_t);
+
+        case NETOPT_CSMA_MINBE:
+            assert(len >= sizeof(uint8_t));
+            *((uint8_t *)value) = dev->csma_min_be;
+            return sizeof(uint8_t);
+
+        case NETOPT_RETRANS:
+            assert(len >= sizeof(uint8_t));
+            *((uint8_t *)value) = dev->max_retrans;
+            return sizeof(uint8_t);
+
+        case NETOPT_TX_RETRIES_NEEDED:
+            assert(len >= sizeof(uint8_t));
+            *((uint8_t *)value) = dev->num_retrans;
+            return sizeof(uint8_t);
+
+        case NETOPT_CHANNEL_PAGE:
+            assert(len >= sizeof(uint16_t));
+            *((uint16_t *)value) = 0;
+            return sizeof(uint16_t);
+
```

Comment:
Might as well drop this if the radio does not support multiple channel pages.

Reply:
Isn't this the only way for upper layers to know that the radio is currently tuned to page 0 as opposed to page 1 or 2? Or is page 0 actually implied by not supporting the request?